src/client/components/form-editor/index.js

```diff
@@ -140,8 +151,84 @@ class FormEditor extends DataComponent {
         let handleElCreation = el => Promise.try( () => synch(el) ).then( () => create(el) );
         let els = [ intn, ...ppts ];
 
-        return Promise.all( els.map(handleElCreation) ).then( () => {
-          return Promise.all( ppts.map(add) ).then( () => add(intn) );
+        let runLayout = cy => {
+          let layout = cy.layout( _.assign( {}, getCyLayoutOpts(), {
+            animate: false,
+            randomize: true
+          } ) );
+
+          // if there is an existing layout disrupt it
+          if ( this.lastLayout ) {
+            this.layoutDisrupted = true;
+            this.lastLayout.stop();
+          }
+
+          this.lastLayout = layout;
+
+          let layoutDone = layout.promiseOn('layoutstop');
+
+          layout.run();
+
+          return layoutDone;
+        };
+
+        let updatePos = el => {
+          let cyEle = cy.getElementById( el.id() );
+          let pos =  _.clone( cyEle.position() );
+
+          el.reposition( pos );
+        };
```

Comment:
At some point in the future, the reposition step could be coalesced into the creation step to reduce the number of ops for performance.  It's relatively fast right now, so let's consider this change later.